src/ConfirmProvider/ConfirmProvider.types.ts

```diff
@@ -1,6 +1,45 @@
-export type ConfirmProviderProps = {
-  /**
-   * The content of the dialog. Valid react element can be used.
-   */
-  children: React.ReactNode;
+import { ButtonProps } from "@mui/material/Button";
+import { DialogActionsProps } from "@mui/material/DialogActions";
+import { DialogContentProps } from "@mui/material/DialogContent";
+import { DialogProps } from "@mui/material/Dialog";
+import { DialogTitleProps } from "@mui/material/DialogTitle";
+import { ReactNode } from "react";
+import { TextFieldProps } from "@mui/material/TextField";
+
+interface ConfirmationDialogOptions {
+  title?: React.ReactNode;
+  description?: React.ReactNode;
+  content?: React.ReactNode;
+  confirmationText?: React.ReactNode;
+  cancellationText?: React.ReactNode;
+  dialogProps?: Partial<DialogProps>;
+  dialogActionsProps?: DialogActionsProps;
+  confirmationButtonProps?: ButtonProps;
+  cancellationButtonProps?: ButtonProps;
+  titleProps?: DialogTitleProps;
+  contentProps?: DialogContentProps;
+  allowClose?: boolean;
+  confirmationKeyword?: string;
+  confirmationKeywordTextFieldProps?: TextFieldProps;
+  hideCancelButton?: boolean;
+  buttonOrder?: Array<"confirm" | "cancel">;
+}
```

Comment:
I've had a look through the VIRTO codebase at how we are using this hook and I can see the following props being used.

```
title
description
confimationText
cancellationText
content
confirmationButtonProps - keep for now (and we can check with Sowbhagya if there is a use case for red buttons)
dialogProps - delete for defaults
titleProps - delete for defaults
cancellationButtonProps - delete for defaults
dialogActionProps - delete for defaults
```

I can't see a valid use case for keeping some of them (they should just use the RUI defaults). The only one that stands out is the confirmationButtonProps where in some apps we show red confirmation buttons. Suggest we keep this for now and get feedback from Sowbhagya later.

The rest can be removed I think and we can simplify the component and stories as a result.